### PR TITLE
fix: Use anonymous github.com creds for API calls on certain repos when there are no github.com credentials available

### LINKED ISSUE #6315: jx boot verify-jenkins-x-environment step fail on github enterprise
### Summary 

log
```
STEP: verify-jenkins-x-environment command: /bin/sh -c jx step verify env in dir: /Users/user/workspace/kubernetes/jenkins-x-boot-config

Storing the requirements in team settings in the dev environment
Validating git repository for dev environment at URL https://[my-github-enterprise-url]/[my-github-enterprise-user]/environment-[my-clusterName]-dev.git
error: handle dev environment repository: creating remote for dev environment environment-[my-clusterName]-dev: duplicating https://github.com/jenkins-x/jenkins-x-boot-config to [my-github-enterprise-user]/environment-[my-clusterName]-dev: getting repo for https://github.com/jenkins-x/jenkins-x-boot-config: failed to get repository jenkins-x/jenkins-x-boot-config due to: GET https://[my-github-enterprise]/api/v3/repos/jenkins-x/jenkins-x-boot-config: 404 Not Found []
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step verify env' command in directory '.', output: ''
```

jx-requirements.yml
```
autoUpdate:
  enabled: false
  schedule: ""
bootConfigURL: https://github.com/jenkins-x/jenkins-x-boot-config.git
cluster:
  clusterName: [my-clusterName]
  environmentGitOwner: [my-github-enterprise-user]
  environmentGitPublic: true
  gitKind: github
  gitName: ghe
  gitPublic: true
  gitServer: https://[my-github-enterprise-url]
  namespace: jx
  provider: kubernetes
  registry: https://[my-harbor-url]/
environments:
- ingress:
    domain: [my-domain]
    externalDNS: true
    namespaceSubDomain: -jx.
    tls:
      email: ""
      enabled: false
      production: false
  key: dev
- ingress:
    domain: ""
    externalDNS: false
    namespaceSubDomain: ""
    tls:
      email: ""
      enabled: false
      production: false
  key: staging
- ingress:
    domain: ""
    externalDNS: false
    namespaceSubDomain: ""
    tls:
      email: ""
      enabled: false
      production: false
  key: production
gitops: true
ingress:
  domain: [my-domain]
  externalDNS: true
  namespaceSubDomain: -jx.
  tls:
    email: ""
    enabled: false
    production: false
kaniko: true
repository: nexus
secretStorage: local
storage:
  backup:
    enabled: false
    url: ""
  logs:
    enabled: false
    url: ""
  reports:
    enabled: false
    url: ""
  repository:
    enabled: false
    url: ""
vault: {}
velero: {}
versionStream:
  ref: v1.0.248
  url: https://github.com/jenkins-x/jenkins-x-versions.git
webhook: lighthouse

```

### Steps to reproduce the behavior

verify-jenkins-x-environment

### Expected behavior

success to get repository jenkins-x/jenkins-x-boot-config from https://github.com (not from [my-github-enterprise])

### Actual behavior

jx boot tries to find a repo in [my-github-enterprise]

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.1065
Kubernetes cluster v1.13.5
kubectl            v1.16.3
helm client        Client: v2.14.3+g0e7f3b6
git                2.20.1 (Apple Git-117)
Operating System   Mac OS X 10.14.6 build 18G87
```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

custom kubernetes (created and provided by our company for private)

### Operating system / Environment

Mac OS X 10.14.6 build 18G87

### LINKED ISSUE #6471: `jx get quickstarts` in a boot gitops cluster with non-github.com fails
If you run `jx get quickstarts` in a boot-created gitops-enabled cluster using anything but `github.com` for your environment/repos/etc, you'll get an error like

```
error: failed to load quickstarts: failed to load quickstarts: creating user for server "https://github.com": running in batch mode and no default API token found
```

This is due to the logic for fetching quickstarts in a gitops environment trying to use an authenticated `GitHubProvider` when there are no GitHub creds available. 

At least for default quickstarts, we should fall back on unauthenticated requests.

Discovered in the process of fixing #6315.

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

There are a few places where we use an authenticated `GitProvider` to perform API calls against github.com repos or orgs we know are public - `DuplicateGitRepoFromCommitish` for cloning/squashing `jenkins-x-boot-config`, and `jx get quickstarts` getting the default quickstarts from the `jenkins-x-quickstarts` org currently. If a boot cluster with gitops enabled is not using github.com and is instead using GitHub Enterprise, Gitlab, etc, these calls end up failing due to trying to use an authenticated provider when there are no credentials available. In these cases, we should fall back to anonymous authentication for the API calls.

#### Special notes for the reviewer(s)

As a followup, once this has merged, released, and gotten into the version stream, I'm going to be updating the Lighthouse and `jenkins-x-versions` BDD tests for GHE/Gitlab/Bitbucket Server to do gitops, so that we don't miss this regressing.

#### Which issue this PR fixes

fixes #6315 
fixes #6471

### CODE REVIEW

**Comment by abayer** [comment]
Well, that's interesting - the LH quickstart label tests are failing:

```
STEP: creating an issue and assigning it to a valid user
      created issue with number 2
      create issue comment on issue 2
STEP: attempting to LGTM our own PR
STEP: adding a hold label
STEP: adding a WIP label

• Failure [936.255 seconds]
quickstart golang-http

/tmp/jenkins-x/bdd-jx/test/suite/quickstart/helpers.go:54
  Create a quickstart
  /tmp/jenkins-x/bdd-jx/test/suite/quickstart/helpers.go:74
    by running jx create quickstart golang-http
    /tmp/jenkins-x/bdd-jx/test/suite/quickstart/helpers.go:75
      creates a new source repository and promotes it to staging [It]
      /tmp/jenkins-x/bdd-jx/test/suite/quickstart/helpers.go:76

      Unexpected error:
          <*errors.errorString | 0xc0006ce080>: {
              s: "the pull request does not have the specified label: do-not-merge/work-in-progress",
          }
          the pull request does not have the specified label: do-not-merge/work-in-progress
      occurred

      /tmp/jenkins-x/bdd-jx/test/suite/quickstart/helpers.go:134
```

**Comment by abayer** [comment]
So `boot-lh-ghe` (just added!) fails here because it turns out the `work-in-progress` label can't be applied with lighthouse on anything but github.com. I'll fix that over at https://github.com/jenkins-x/lighthouse/pull/452 eventually, but there's no need to block this here any longer.

File: pkg/gits/helpers.go [hunk 1]
```diff
@@ -776,7 +777,7 @@ func FindTagForVersion(dir string, version string, gitter Gitter) (string, error
 // DuplicateGitRepoFromCommitish will duplicate branches (but not tags) from fromGitURL to toOrg/toName. It will reset the
 // head of the toBranch on the duplicated repo to fromCommitish. It returns the GitRepository for the duplicated repo.
 // If the repository already exist and error is returned.
-func DuplicateGitRepoFromCommitish(toOrg string, toName string, fromGitURL string, fromCommitish string, toBranch string, private bool, provider GitProvider, gitter Gitter) (*GitRepository, error) {
+func DuplicateGitRepoFromCommitish(toOrg string, toName string, fromGitURL string, fromCommitish string, toBranch string, private bool, provider GitProvider, gitter Gitter, fromProvider GitProvider) (*GitRepository, error) {
```
Comment (hferentschik) [inline pkg/gits/helpers.go]: damn you @abayer ;-) I am making changes to the same files/functions. Guess I will have to see how I can consolidate later

**Comment by hferentschik** [comment]
@abayer where are we with this? Needs another lgtm?

**Pull request merged**